op-program/client/l1/oracle.go

```diff
@@ -86,3 +91,31 @@ func (p *PreimageOracle) ReceiptsByBlockHash(blockHash common.Hash) (eth.BlockIn
 
 	return info, receipts
 }
+
+func (p *PreimageOracle) GetBlob(ref eth.L1BlockRef, blobHash eth.IndexedBlobHash) *eth.Blob {
+	indexBytes := make([]byte, 32)
+	binary.BigEndian.PutUint64(indexBytes, blobHash.Index)
+
+	// Hint for the commitment, the preimage of the blob versioned hash.
+	p.hint.Hint(Sha2Hint(blobHash.Hash))
+	commitment := p.oracle.Get(preimage.Sha256Key(blobHash.Hash))
+
+	// Send a hint for the blob field elements.
+	blobHint := make([]byte, 32)
+	binary.BigEndian.PutUint64(blobHint[8:16], blobHash.Index)
+	binary.BigEndian.PutUint64(blobHint[24:32], ref.Time)
+	p.hint.Hint(BlobHint(blobHint))
+
+	// Reconstruct the full blob from the 4096 field elements.
+	blob := eth.Blob{}
+	fieldElemKey := make([]byte, 80)
+	copy(fieldElemKey[:48], commitment)
+	for i := 0; i < 4096; i++ {
```

Comment:
There are probably constants in the `eth` package and/or the geth `params` package for the points-per-blob.